rust/arrow/src/array/array.rs

```diff
@@ -1309,6 +1309,11 @@ impl StringArray {
     fn value_offset_at(&self, i: usize) -> i32 {
         unsafe { *self.value_offsets.get().add(i) }
     }
+
```

Comment:
For consistency, may you please also add `builder` to `BinaaryArray`. They both behave the same way